### PR TITLE
[merged] Fetcher cleanup

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

**Comment by cgwalters** [comment]
OK, this one should be good to review and land I think.

File: src/libostree/ostree-fetcher.c [hunk 1]
```diff
@@ -1233,33 +1266,64 @@ _ostree_fetcher_request_finish (OstreeFetcher         *self,
...
+{
+  _ostree_fetcher_request_async (self, mirrorlist, filename, flags, TRUE,
+                                 max_size, priority, cancellable,
+                                 callback, user_data);
+}
+
+gboolean
+_ostree_fetcher_request_to_membuf_finish (OstreeFetcher *self,
+                                          GAsyncResult  *result,
+                                          GBytes       **out_filename,
```
Comment (jlebon) [inline src/libostree/ostree-fetcher.c]: `out_bytes` ?

File: src/libostree/ostree-fetcher.h [hunk 2]
```diff
@@ -103,25 +103,39 @@ void _ostree_fetcher_set_extra_headers (OstreeFetcher *self,
...
+                                        OstreeFetcherRequestFlags flags,
+                                        guint64                max_size,
+                                        int                    priority,
+                                        GCancellable          *cancellable,
+                                        GAsyncReadyCallback    callback,
+                                        gpointer               user_data);
+
+gboolean _ostree_fetcher_request_to_membuf_finish (OstreeFetcher *self,
+                                                   GAsyncResult  *result,
+                                                   GBytes       **out_filename,
```
Comment (jlebon) [inline src/libostree/ostree-fetcher.h]: `out_bytes` ?

**Comment by cgwalters** [comment]
:ballot_box_with_check:

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 78609ca

**Comment by jlebon** [comment]
@rh-atomic-bot r+ ffcc250

**Pull request closed without merging**